src/nvim/main.c

```diff
@@ -186,6 +183,12 @@ int nvim_main(int argc, char **argv)
 int main(int argc, char **argv)
 #endif
 {
+  argv0 = basename(argv[0]);
```

Comment:
@oni-link I'll respond to your comment here as well (so it doesn't disappear if I amend the commit).

> would be more OS-independent

Could you explain why? I don't know much about libgen's portability, but it's specified in POSIX (although that doesn't necessarily mean Windows has it :p).

> Also, never returns NULL, but will return an empty string if argv[0] == NULL

So does basename, at least I think so after reading the [POSIX article on it](http://pubs.opengroup.org/onlinepubs/9699919799/functions/basename.html). I'm checking for null because the implementation I'm using (OpenBSD) mentions this:
```
ERRORS
     The following error codes may be set in errno:

     [ENAMETOOLONG]     The path component to be returned was larger than
                        PATH_MAX.
```

> Is that even possible?

I've no idea... maybe if nvim was executed using `exec`?